client/src/components/ActionMenu/ActionMenu.js

```diff
@@ -20,9 +20,9 @@ class ActionMenu extends PureComponent {
       toggleCallback(event);
     }
 
-    this.setState({
-      isOpen: !this.state.isOpen
-    });
+    // Force setting state to the end of the execution queue to clear a potential race condition
+    // with entwine click handlers
+    window.setTimeout(() => this.setState({ isOpen: !this.state.isOpen }), 0);
```

Comment:
I've added this in a couple of places defensively - the only place I _know for certain_ it's needed is this one.
The race condition occurs like this:
1. User clicks the button
2. React responds first.
    1. React toggles the menu to close, which _removes_ the button that was clicked
3. Entwine wants to respond to the event, but the button isn't in the DOM so it doesn't.

This solution works because it defers closing the dropdown (and thus removing the button) until the _next event cycle_ (see [the MDN docs for `setTimeout`](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#delay)). This allows the current event cycle to continue (and therefore lets entwine respond to the click event) before closing closing the dropdown.
More about the event loop can be read in [MDN's docs on the subject](https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop).